### ISSUE TITLE
eslint isn't totally disabled when Prettier is integrated with it

### ISSUE THREAD

**Issue by rolandleth** [wallabyjs/console-ninja#4]
Not sure how this can be fixed, but when using Prettier in ESLint:

<img width="1527" alt="2022-12-08 at 12 05 45" src="https://user-images.githubusercontent.com/1858700/206419044-b1153cc6-361f-42cf-84db-0da84053b912.png">
<img width="1686" alt="Screenshot 2022-12-08 at 12 05 50" src="https://user-images.githubusercontent.com/1858700/206418905-5bc8f078-5a02-4a69-bf0e-b309ddac0b49.png">

The only differences between the code and Prettier suggestion (just at the start and end):

<img width="1560" alt="2022-12-08 at 12 08 31" src="https://user-images.githubusercontent.com/1858700/206419481-12fe89e7-e217-4ae8-811b-6b7f026e8c60.png">
<img width="1555" alt="Screenshot 2022-12-08 at 12 08 36" src="https://user-images.githubusercontent.com/1858700/206419452-7c8b9fe7-96ac-4b11-a84b-fad4696e8895.png">

Adding an extra `// eslint-disable` above the `console.log` doesn't fix this either 🤔 I can only make the error go away by adding `// eslint-disable-next-line prettier/prettier` to the line above.

**Comment by rolandleth** [comment]
Hmm, I think I found the problem (had a project where I had no issues, so I just diff'd them).

```
"parserOptions": {
        // ...
	"ecmaFeatures": {
		"jsx": true,
		"impliedStrict": true
	}
},
```

The one with issues uses `impliedStrict: true` and if I comment it out, the issue goes away. I'm not a web developer at core, so I don't even know if this is _actually_ needed or not 🤔

**Comment by ArtemGovorov** [comment]
Thanks for the update. We are happy to investigate why exactly it was failing if you can provide us the info requested in my previous post: https://github.com/wallabyjs/console-ninja/issues/4#issuecomment-1343773689.

**Comment by rolandleth** [comment]
Sure, I just assumed that by pointing out the issue it'd be enough, because it can be replicated just by itself in a bare `create-react-app` project:

package.json:
```json
{
  "name": "example",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "devDependencies": {
    "eslint": "^8.29.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-prettier": "^4.2.1",
    "prettier": "^2.8.1"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

```

eslint:
```json
{
	"extends": [
		"eslint:recommended",
		"plugin:react/recommended",
		"plugin:jest/recommended",
		"plugin:react-hooks/recommended",
		"plugin:react/jsx-runtime",
		"plugin:prettier/recommended"
	],
	"plugins": ["prettier"],
	"parserOptions": {
		"sourceType": "module",
		"warnOnUnsupportedTypeScriptVersion": false,
		"ecmaFeatures": {
			"jsx": true
		}
	},
	"env": {
		"es2020": true,
		"jest": true
	},
	"ignorePatterns": [
		".next",
		"node_modules",
		"build",
		"coverage",
		"*.config.js"
	],
	"rules": {
		"no-unused-vars": "off",
		"no-empty": ["error", { "allowEmptyCatch": true }]
	},
	"settings": {
		"react": {
			"version": "detect"
		}
	}
}
```

prettierrc:
```
```

**Comment by NikGovorov** [comment]
Thanks for reporting the issue. It is fixed in v0.0.34.

**Comment by rolandleth** [comment]
Can confirm it's fixed @NikGovorov, but it broke in two other places (might be related, tho): 

1. If I have a rule for, let's say `@typescript-eslint/no-empty-function`, which I disable at the start of the file, any `console.logs` placed before any empty function in that file will make the `eslint-disable` at the very top not work and errors about empty functions will start popping up. If all the `console.logs` are after _all_ empty functions, everything works properly.
2. I also have `"import/exports-last": "warn",` and any `console.log` within a file will make that warning pop up, even if I have a single `export default` at the end of the file. Removing the `console.log` will make the warning go away.

Of course, if I disable Console Ninja, none of these problems exist.

**Comment by NikGovorov** [comment]
Thanks for reporting the issue. It is fixed in v0.0.41.

**Comment by rolandleth** [comment]
Can confirm it's fixed, thanks for the quick fix! 💯

**Comment by rolandleth** [comment]
Hey, @NikGovorov, this started happening again, but when using breakpoints or logpoints.

**Comment by ArtemGovorov** [comment]
@rolandleth Can you please share a screenshot/text of the error that is happening using breakpoints or logpoints?

**Comment by rolandleth** [comment]
Sure:
<img width="988" alt="Screenshot 2023-03-08 at 12 47 21" src="https://user-images.githubusercontent.com/1858700/223693145-eb5c3ed9-0795-46a6-b8a2-f180f84b0f16.png">

Where line 99 is this:
<img width="538" alt="Screenshot 2023-03-08 at 12 48 05" src="https://user-images.githubusercontent.com/1858700/223693327-0b57a432-cf3e-41bc-b03b-f56c51fe351d.png">

There is no `console.log` in this file.

**Comment by NikGovorov** [comment]
We were not able to reproduce the issue, could you please try https://github.com/wallabyjs/cn-issue-4 App.tsx:7? If the repo works for you, please fork and change it in a way that replicates the rules from your real project.

**Comment by rolandleth** [comment]
Sure, changed `App.tsx` to this:

```ts
function App() {
	let msg1 = "hello typescript"
	const counter = 1
	const flag1 = true
	if (msg1 === "hello typescript" && flag1) {
		msg1 = "hello javascript"
	}
	return (
		<div className="App">
			<header className="App-header">
				<img
					src={
						((counter > 1 || counter > 5) && counter > 3) ||
						(flag1 && msg1 == "hi")
							? logo
							: logo1
					}
					className="App-logo"
					alt="logo"
				/>
				<p>
					Edit <code>src/App.tsx</code> and save to reload.
				</p>
				<a
					className="App-link"
					href="https://reactjs.org"
					target="_blank"
					rel="noopener noreferrer"
				>
					Learn React
				</a>
			</header>
		</div>
	)
}

export default App
```

**Comment by NikGovorov** [comment]
Thanks for reporting the issue. It is fixed in v0.0.82.
